STYLE_GUIDE.md

```diff
@@ -0,0 +1,164 @@
+# Code Aesthetics
+## Summary
+This document is in addition to the [BSG SystemVerilog Coding Guidelines](https://docs.google.com/document/d/1xA5XUzBtz_D6aSyIBQUwFk_kSUdckrfxa2uzGjMgmCU). All rules in that file should be obeyed. This document augments that documents with a few aesthetic guidelines.
+This document is intended to provide more rigid structure for coding style and file structure.
+
+## References
+[BSG SystemVerilog Coding Guidelines](https://docs.google.com/document/d/1xA5XUzBtz_D6aSyIBQUwFk_kSUdckrfxa2uzGjMgmCU)
+
+[Freescale Verilog Guidelines](https://people.ece.cornell.edu/land/courses/ece5760/Verilog/FreescaleVerilog.pdf)
+
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
+
+[lowRISCV Verilog Coding Style Guide](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md)
+
+## Verilog Style
+- No tabs in files, anywhere.  People get different results based on their editor.
+    - Exception: Makefiles. Tabs should be used to indent recipes, but not for general alignment
+- Align all things that have parallel structure (begin/end, case/endcase, commas, periods, etc.). This makes it easier to check the code and find the matching component.
+
+        if (x_p)  // notice how matching inputs are aligned
+          begin: foo
+            adder     (this_input, that_input, another_input);
+          end
+        else
+          begin: bar
+            wow_adder (this_input,          3, another_input);
+          end
+    
+    
+        if (foo)    // notice alignment of if/else, begin/end
+            x = 3;  // and optional alignment of x = statement.
+        else
+          begin
+            x = 1;
+	    	y = 0;
+          end
+
+- Use localparam keyword for derived parameters and input parameters that should not be set by the party instantiating the module.
+- Localparams in port lists are allowed, but only for sizing ports. Other generated localparams should go near their consumers.
+- Prefer input [1:0] to input logic [1:0] and output [1:0] to output logic [1:0]
```

Comment:
I'm not sure we follow this. Most files use "output logic", I think.

Reply:
Yeahh, it's definitely not a well-followed rule (Prefer vs Use).  I think it's still valuable for new code, but I added in some more justification.